slomonitor/src/main/scala/com.gu.notifications.slos/SloMonitor.scala

```diff
@@ -0,0 +1,68 @@
+package com.gu.notifications.slos
+
+import com.amazonaws.auth.{AWSCredentialsProviderChain, DefaultAWSCredentialsProviderChain}
+import com.amazonaws.auth.profile.ProfileCredentialsProvider
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.athena.AmazonAthenaAsyncClient
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.gu.notifications.athena.{Athena, Query}
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.time.{Instant, LocalDateTime, ZoneOffset}
+import java.util.concurrent.{Executors, ScheduledExecutorService, TimeUnit}
+import scala.concurrent.{Await, ExecutionContext, duration}
+object SloMonitor {
+
+  import ExecutionContext.Implicits.global
+
+  val logger: Logger = LoggerFactory.getLogger(this.getClass)
+
+  val stage: String = System.getenv("STAGE")
+
+  val credentials = new AWSCredentialsProviderChain(
+    new ProfileCredentialsProvider("mobile"),
+    DefaultAWSCredentialsProviderChain.getInstance
+  )
+
+  implicit val athenaClient = AmazonAthenaAsyncClient.asyncBuilder()
+    .withCredentials(credentials)
+    .withRegion(Regions.EU_WEST_1)
+    .build()
+
+  implicit val scheduledExecutorService: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
+
+  def calculatePercentage(actualDeliveries: Double, expectedDeliveries: Double): Double = {
+    (actualDeliveries / expectedDeliveries) * 100
+  }
+
+  def handleMessage(event: SQSEvent): Unit = {
+    logger.info(s"Running SLO monitor")
+    val record = event.getRecords.get(0) // Batch size is defined as 1 in cdk
+    val notificationId = record.getBody
+    val sentTime: LocalDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(record.getAttributes.get("SentTimestamp").toLong), ZoneOffset.UTC)
+    // The number of expected deliveries is hardcoded for the purposes of this prototype, but we could:
+    // a) Send this as part of the SQS message (from the notifications app)
+    // b) Obtain it via the DynamoDB Report table
```

Comment:
I originally wrote the code for b, but it relied on the `common` package, which made the `.jar` too large, so I've reverted that work for now!